### PR TITLE
Install downstream packages in correct `subdir`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Xref https://github.com/conda/conda-build/issues/4750

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda_build/render.py [hunk 1]
```diff
@@ -217,9 +218,18 @@ def find_pkg_dir_or_file_in_pkgs_dirs(pkg_dist, m, files_only=False):
...
+            try:
+                info = json.loads(Path(pkg_dir, "info", "index.json").read_text())
+                subdir = info["subdir"]
+            except (FileNotFoundError, KeyError):
+                subdir = m.config.host_subdir
+
+            pkg_subdir = os.path.join(m.config.croot, subdir)
             pkg_loc = os.path.join(pkg_subdir, os.path.basename(pkg_file))
             shutil.move(pkg_file, pkg_loc)
+            break
```
Comment (dholth) [inline conda_build/render.py:232]: why break?

File: conda_build/render.py [hunk 2]
```diff
@@ -217,9 +218,18 @@ def find_pkg_dir_or_file_in_pkgs_dirs(pkg_dist, m, files_only=False):
...
                 for entry in os.listdir(pkg_dir):
                     archive.add(os.path.join(pkg_dir, entry), arcname=entry)
-            pkg_subdir = os.path.join(m.config.croot, m.config.host_subdir)
+
+            # use the package's subdir
+            try:
+                info = json.loads(Path(pkg_dir, "info", "index.json").read_text())
+                subdir = info["subdir"]
+            except (FileNotFoundError, KeyError):
+                subdir = m.config.host_subdir
```
Comment (kenodegard) [inline conda_build/render.py:227]: I dislike this fallback, what do we do instead? let the errors propagate up?

Comment (dholth) [reply conda_build/render.py:227]: old packages don't have subdir IIRC

**Review by dholth — APPROVED** [review]
LGTM

**Review by jezdez — APPROVED** [review]

**Comment by jezdez** [comment]
This had missed news and tests :-/

**Pull request merged**